addons/io/org.openhab.io.hueemulation/src/main/java/org/openhab/io/hueemulation/internal/HueEmulationServlet.java

```diff
@@ -344,13 +374,16 @@ private void apiState(String id, HttpServletRequest req, HttpServletResponse res
      */
     private void apiLight(String id, HttpServletRequest req, HttpServletResponse resp) throws IOException {
         try {
-            Item item = itemRegistry.getItem(id);
+            HueDevice device = getHueDevices().get(new Integer(id));
+            if (device == null) {
```

Comment:
You rather not throw the class exception directly, consider using a more describing one or a custom one, maybe illegal state exception fits here? And can you then please change the catch so it does not catch all exceptions but just specific ones.